nix/crossBuild.nix

```diff
@@ -0,0 +1,85 @@
+{ nixpkgs, system, version }:
+let
+  # ld flag to embed the version in the binary
+  ldEmbedVersion =
+    "-X github.com/dividat/driver/src/dividat-driver/server.version=${version}";
+
+  buildGoModuleFor = targetSystem: customAttrs:
+    let
+      crossPkgs = (import nixpkgs {
+        system = system;
+        crossSystem.config = targetSystem;
+      });
+    in
+    with crossPkgs;
+    buildGoModule ({
+      pname = "dividat-driver";
+      version = version;
+      vendorHash = "sha256-S8qJorySFPERNp4i9ckeZCPLyx0d/QweS4hm4ghJT4k=";
+      ldflags = [
+        ldEmbedVersion
+        "-linkmode external"
+        "-extldflags '-static'"
```

Comment:
Is this Linux-specific or does it apply to Windows too?

Reply:
Kind of an outdated answer, but this suggests static linking is also possible for windows builds:
https://stackoverflow.com/questions/43311621/c-correct-way-to-statically-dynamically-link-with-mingw-w64

Reply:
I would go with dynamic linking for Windows, it's worked well so far.

Reply:
The windows build was also statically linked previously, if I'm not mistaken:
https://github.com/dividat/driver/blob/2aaf5787864a199a928404f3bea8cc659de52022/Makefile#L77

https://github.com/dividat/driver/blob/2aaf5787864a199a928404f3bea8cc659de52022/Makefile#L29-L43

I'm not sure if it's really necessary though.